From 993061fbb3cb4b2c3f3a2621ef6f18a53e74e55c Mon Sep 17 00:00:00 2001
From: Devi Priya <quic_devipriy@quicinc.com>
Date: Wed, 27 Dec 2023 15:31:28 +0530
Subject: [PATCH] soc: qcom: llcc: Add configuration data for IPQ54xx

Add LLCC configuration data for IPQ54xx SoC which
controls the LLCC behaviour.

Change-Id: Ibb5218271e33ebbfa5f6c4e20e8369e488ac3ff0
Signed-off-by: Devi Priya <quic_devipriy@quicinc.com>
---
 drivers/soc/qcom/llcc-qcom.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/drivers/soc/qcom/llcc-qcom.c b/drivers/soc/qcom/llcc-qcom.c
index c61848595da0..f84eced8a292 100644
--- a/drivers/soc/qcom/llcc-qcom.c
+++ b/drivers/soc/qcom/llcc-qcom.c
@@ -356,6 +356,13 @@ static const struct llcc_slice_config sm8550_data[] =  {
 	{LLCC_VIDVSP,   28,  256, 4, 1, 0xFFFFFF, 0x0,   0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, },
 };
 
+static const struct llcc_slice_config devsoc_data[] =  {
+	/* SCID 1 - Descriptors (PPE, CUMAC) */
+	{LLCC_CPUSS,     1,  768, 1, 0,  0xFFFF, 0x0,  0, 0, 0, 1, 1, 0, 1, 1, 1, 0, 0, 1, 1, 1, 1, },
+	/* SCID 2 - SKB, SKB data (PPE, CUMAC, APSS Core 0,1,2) */
+	{LLCC_VIDSC0,    2,  256, 2, 1,  0xF000, 0x0,  0, 0, 0, 1, 1, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, },
+};
+
 static const struct llcc_edac_reg_offset llcc_v1_edac_reg_offset = {
 	.trp_ecc_error_status0 = 0x20344,
 	.trp_ecc_error_status1 = 0x20348,
@@ -519,6 +526,14 @@ static const struct qcom_llcc_config sm8550_cfg = {
 	.edac_reg_offset = &llcc_v2_1_edac_reg_offset,
 };
 
+static const struct qcom_llcc_config devsoc_cfg = {
+	.sct_data       = devsoc_data,
+	.size           = ARRAY_SIZE(devsoc_data),
+	.need_llcc_cfg	= true,
+	.reg_offset	= llcc_v2_1_reg_offset,
+	.edac_reg_offset = &llcc_v2_1_edac_reg_offset,
+};
+
 static struct llcc_drv_data *drv_data = (void *) -EPROBE_DEFER;
 
 /**
@@ -1053,6 +1068,7 @@ static int qcom_llcc_probe(struct platform_device *pdev)
 }
 
 static const struct of_device_id qcom_llcc_of_match[] = {
+	{ .compatible = "qcom,devsoc-llcc", .data = &devsoc_cfg },
 	{ .compatible = "qcom,sc7180-llcc", .data = &sc7180_cfg },
 	{ .compatible = "qcom,sc7280-llcc", .data = &sc7280_cfg },
 	{ .compatible = "qcom,sc8180x-llcc", .data = &sc8180x_cfg },
-- 
2.34.1

